### PR TITLE
Add test fixtures and initial integration test suite

### PR DESCRIPTION
**Pull request by ivantha**
## Summary

Adds test infrastructure with pre-generated fixtures and a first integration test suite.

### New Files
- \`tests/fixtures/tasks.json\` — 200 sample tasks (generated)
- \`tests/fixtures/users.json\` — 50 sample users (generated)
- \`tests/fixtures/README.md\` — Fixture documentation
- \`tests/setup.ts\` — Test helpers (seeding, auth tokens)
- \`tests/tasks.test.ts\` — Integration tests for Tasks API

### Test Coverage
- GET /api/tasks (pagination, filtering by status/priority, auth)
- POST /api/tasks (creation, validation)
- PATCH /api/tasks/:id (updates, 404)
- DELETE /api/tasks/:id

## Stats
~3500 lines added (mostly fixture JSON data)

## Note
This is a large PR due to the fixture data. The actual test code is ~120 lines.

### CODE REVIEW

File: tests/fixtures/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Test Fixtures
```
Comment (ivantha) [inline tests/fixtures/README.md:1]: **[INFO]** Fixture README could be more descriptive

The `tests/fixtures/README.md` file provides a good overview of the fixtures and their regeneration process. However, it could benefit from more specific details about the fields within `tasks.json` and `users.json`, such as the expected data types or any constraints on values (e.g., possible statuses for tasks).

**Suggestion:**
```
Expand the 'Files' section to include brief descriptions of the schema for `tasks.json` and `users.json`. For example:

- `tasks.json` — 200 sample tasks with varied statuses (e.g., 'todo', 'in_progress', 'done', 'cancelled'), priorities ('low', 'medium', 'high'), tags, and dates.
- `users.json` — 50 sample users with different roles ('member', 'viewer', 'admin').
```

File: tests/setup.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export function seedDatabase(): void {
+  for (const task of tasks) {
+    db.createTask(task as any);
+  }
+  for (const user of users) {
+    db.createUser({
+      ...user,
+      passwordHash: "$2b$10$placeholder-hash-for-testing",
+    } as any);
```
Comment (ivantha) [inline tests/setup.ts:13]: **[INFO]** In `tests/setup.ts`, `clearDatabase` is a placeholder.

The `clearDatabase` function in `tests/setup.ts` is currently a placeholder with a TODO comment. For a robust test suite, especially when tests might modify data, a proper implementation to clear or reset the database state between test runs (or suites) is crucial to ensure test isolation and prevent state leakage.

**Suggestion:**
```
Implement a `clear()` method in the `Database` class (or wherever `db` is defined) and call it here. If direct clearing is not feasible, consider strategies like using in-memory databases for tests or leveraging database snapshots if applicable.
```

File: tests/setup.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { db } from "../src/db";
+import tasks from "./fixtures/tasks.json";
+import users from "./fixtures/users.json";
+
+export function seedDatabase(): void {
```
Comment (ivantha) [inline tests/setup.ts:5]: **[INFO]** Type assertion `as any` used for database seeding.

In `tests/setup.ts`, type assertions like `task as any` and `user as any` are used when seeding the database. While this might work for testing purposes, it bypasses TypeScript's type checking and can hide potential mismatches between the fixture data structure and the expected types for `db.createTask` and `db.createUser`. This can lead to runtime errors if the fixture data does not conform to the expected shape.

**Suggestion:**
```
Define interfaces or types that match the structure of `tasks.json` and `users.json` and ensure they are compatible with the arguments expected by `db.createTask` and `db.createUser`. If the structures are indeed different, adjust the seeding logic or the fixture data to align with the types, or create explicit type transformations instead of using `as any`.
```

File: tests/setup.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    db.createUser({
+      ...user,
+      passwordHash: "$2b$10$placeholder-hash-for-testing",
+    } as any);
+  }
+}
+
+export function clearDatabase(): void {
+  // Reset by creating a fresh instance
+  // TODO: Add a clear() method to the Database class
```
Comment (ivantha) [inline tests/setup.ts:19]: **[INFO]** JWT secret hardcoded for testing.

In `tests/setup.ts`, the JWT secret is hardcoded as `'test-secret-for-jest'` for testing purposes. While this is common for test environments, it's good practice to ensure this is clearly documented or managed as an environment variable even in tests to avoid accidental usage in production or sensitive environments.

**Suggestion:**
```
Consider loading the test JWT secret from a dedicated test environment variable (e.g., `process.env.TEST_JWT_SECRET`) and fall back to a default only if necessary. Ensure this test secret is different from any potential production or staging secrets.
```

**Review by ivantha — COMMENTED** [review]
## ◎ Odinro Review

This PR introduces comprehensive test fixtures and an integration test suite for the Tasks API. It includes seeding the database with mock data and tests for CRUD operations and filtering. The tests appear to be well-structured and cover the specified API endpoints effectively.

**Suggested Version Bump:** `patch`

File: tests/tasks.test.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
+import { describe, it, expect, beforeAll } from "@jest/globals";
```
Comment (ivantha) [inline tests/tasks.test.ts:1]: **[INFO]** Integration tests could include more negative test cases.

The integration tests in `tests/tasks.test.ts` cover common success paths and some basic error cases (e.g., missing title, 404 for PATCH/DELETE). However, more negative test cases could be added for robustness.

Examples:
- `POST /api/tasks`: Invalid data types for fields (e.g., status, priority, dueDate), missing required fields not currently checked (like `assigneeId` if it were mandatory), exceeding length limits.
- `PATCH /api/tasks/:id`: Attempting to update with invalid data types, trying to update a read-only field.
- `DELETE /api/tasks/:id`: Attempting to delete a non-existent task.
- Authorization: Testing with tokens of different roles (e.g., 'viewer' trying to perform actions they shouldn't).

**Suggestion:**
```
Add more test cases to cover invalid input, incorrect data types, boundary conditions, and authorization failures for each API endpoint. This will increase the confidence in the API's resilience and security.
```
